frontend/isoko/src/components/business/BusinessPreview.tsx

```diff
@@ -82,15 +84,23 @@ interface BusinessPreviewProps extends React.HTMLProps<HTMLDivElement> {
    verified: boolean;
    path: string;
    numReviews: number;
+   businessId: string;
 }
 
 const BusinessPreview: React.FC<BusinessPreviewProps> = (props) => {
    const navigate = useNavigate();
 
+   const dispatch = useAppDispatch();
+
+   const getClickedBusinessDetails = (businessId) => {
+      dispatch(getBusinessAsync(businessId));
+   };
```

Comment:
Nit: you could also call dispatch in your onClick function since this function is just a single call. But not really a big deal cause this way it's a bit more verbose